plugins/plugin-cpp/che-plugin-cpp-lang-ide/src/main/java/org/eclipse/che/plugin/cpp/ide/action/CreateCppSourceFileAction.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.cpp.ide.action;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.eclipse.che.ide.newresource.AbstractNewResourceAction;
+import org.eclipse.che.plugin.cpp.ide.CppLocalizationConstant;
+import org.eclipse.che.plugin.cpp.ide.CppResources;
+
+import static org.eclipse.che.plugin.cpp.shared.Constants.CPP_EXT;
+
+/**
+ * Action to create new C++ source file.
+ *
+ * @author Vitalii Parfonov
+ */
+@Singleton
+public class CreateCppSourceFileAction extends AbstractNewResourceAction {
```

Comment:
I think that we should show this action only for an appropriate project type(s). But not for all projects.
WDYT?